helm/oauth2-proxy/templates/deployment.yaml

```diff
@@ -147,7 +147,13 @@ spec:
 {{ tpl (toYaml .Values.extraEnv) . | indent 8 }}
         {{- end }}
         ports:
+        {{- if (and (eq .Values.httpScheme "http") (empty .Values.containerPort)) }}
           - containerPort: 4180
+        {{- else if (and (eq .Values.httpScheme "https") (empty .Values.containerPort)) }}
+          - containerPort: 4443
+        {{- else }}
+          - containerPort: {{ .Values.containerPort }}
```

Comment:
Container port should take precedence, I think you cannot have the schema empty, so this else block will never be used.

Reply:
I've changed the order around so containerPort takes precedence. 

This isn't assuming the httpScheme is empty. The logic will set the value of the containerPort to a default value if the httpScheme is set and the containerPort is not. If the containerPort is defined then it will be set to what the user entered in the values.yaml. They will continue to get an error if they don't configure the httpScheme value.

Reply:
@endakelly you are right. I didn't look properly. In that case, you could have left the order as is. Now the checks for empty containerPort are redundant, in the new commit.